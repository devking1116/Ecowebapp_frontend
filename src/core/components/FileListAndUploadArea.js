/* @flow */
import React from 'react';
import reactCSS from 'reactcss';
import { Icon } from 'semantic-ui-react';

type Props = {
  children?: any,
};

/**
 * A file upload area.
 */
export default function FileListAndUploadArea({ children, ...otherProps}: Props): React.Element<*> {
  const styles = reactCSS({
    'default': {
      wrapper: {
        display: 'flex',
        flexDirection: 'column',
        justifyContent: 'space-between',
        height: 'auto',
        margin: 10,
        padding: 5,
        borderWidth: 2,
        borderRadius: 5,
        borderStyle: 'dashed',

        borderColor: '#D4D4D5', //NOTE, from theme
        color: '#D4D4D5',       //NOTE, from theme
      },
      instructions: {
        display: 'flex',
        justifyContent: 'center',
        marginTop: 50,
      },
      browseLink: {
        marginLeft: 5,
      },
      attachments: {
        margin: 10,
        paddingTop: 20,
      },
    },
  });

  return (
    <div {...otherProps}>
      <div style={styles.wrapper}>
        <span style={styles.instructions}>
          <Icon name="cloud upload" size="large" />
          <span>Drop files here to upload, </span>
          <a style={styles.browseLink} href="#"> or browse</a>
        </span>
        <div style={styles.attachments}>
          {children}
        </div>
      </div>
    </div>
  );
}
