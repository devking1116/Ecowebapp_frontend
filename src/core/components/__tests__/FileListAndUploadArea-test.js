import React from 'react';
import renderer from 'react-test-renderer';

import FileListAndUploadArea from '../FileListAndUploadArea';

test('it displays it children at the bottom of the drop area.', () => {
  const component = renderer.create(
    <FileListAndUploadArea />
  );
  const tree = component.toJSON();

  expect(tree).toMatchSnapshot();
});
