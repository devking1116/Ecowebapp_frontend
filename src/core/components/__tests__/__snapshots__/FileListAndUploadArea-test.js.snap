exports[`test it displays it children at the bottom of the drop area. 1`] = `
<div>
  <div
    style={
      Object {
        "MozBorderRadius": 5,
        "OBorderRadius": 5,
        "WebkitBorderRadius": 5,
        "WebkitJustifyContent": "space-between",
        "borderColor": "#D4D4D5",
        "borderRadius": 5,
        "borderStyle": "dashed",
        "borderWidth": 2,
        "color": "#D4D4D5",
        "display": "flex",
        "flexDirection": "column",
        "height": "auto",
        "justifyContent": "space-between",
        "margin": 10,
        "msBorderRadius": 5,
        "padding": 5,
      }
    }>
    <span
      style={
        Object {
          "WebkitJustifyContent": "center",
          "display": "flex",
          "justifyContent": "center",
          "marginTop": 50,
        }
      }>
      <i
        aria-hidden="true"
        className="cloud upload large icon" />
      <span>
        Drop files here to upload, 
      </span>
      <a
        href="#"
        style={
          Object {
            "marginLeft": 5,
          }
        }>
         or browse
      </a>
    </span>
    <div
      style={
        Object {
          "margin": 10,
          "paddingTop": 20,
        }
      } />
  </div>
</div>
`;
